### PR TITLE
feat: Return role & ACLs (permissions) for case type(s) to CCD (AM-423)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/AM-423

Change description
feat: Return role & ACLs (permissions) for case type(s) to CCD (AM-423)

### CODE REVIEW

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java [hunk 1]
```diff
@@ -115,4 +124,12 @@
         + "and (:relationship is null or access_management.relationship = :relationship) "
         + "and (access_management.attribute = :attributeAsString or access_management.attribute like concat(:attributeAsString, '/', '%'))")
     void revokeAccessManagementForAudit(@BindBean ExplicitAccessMetadata explicitAccessMetadata);
+
+    @SqlQuery("select resource_name as caseTypeId, role_name as role, permissions from default_permissions_for_roles "
+        + "where resource_type = 'case' and resource_name in (<caseTypeIds>) "
+        + "and attribute = '' order by resource_name, role_name")
+    @RegisterBeanMapper(RolePermissionsForCaseTypeEnvelope.class)
+    @RegisterBeanMapper(DefaultRolePermissions.class)
```
Comment (shashank-rastogi) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java]: Please move the queries to the defaultroleImportService repository

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/DefaultRolePermissions.java [hunk 2]
```diff
@@ -12,6 +13,7 @@
 @Data
 @Builder
 @AllArgsConstructor
+@NoArgsConstructor //required for JBI Bean Mapper RolePermissionsForCaseTypeEnvelopeReducer
```
Comment (shashank-rastogi) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/DefaultRolePermissions.java]: comment to be corrected - "required for J**D**BI Bean Mapper RolePermissionsForCaseTypeEnvelopeReducer"

**Review by shashank-rastogi — CHANGES_REQUESTED** [review]

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/DefaultRolePermissions.java (hunk 2 above)
Comment (abhijitdiwan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/DefaultRolePermissions.java]: done

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java (hunk 1 above)
Comment (abhijitdiwan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/repositories/AccessManagementRepository.java]: done

**Review by shashank-rastogi — APPROVED** [review]

**Pull request merged**